python/rmm/_lib/device_pointer.pyx

```diff
@@ -0,0 +1,13 @@
+from rmm._lib.lib cimport c_free, cudaStream_t
+
+cdef class DevicePointer:
+    def __cinit__(self, ptr, stream=0):
+        self.c_ptr = <void*><size_t>(ptr)
+        self.c_stream = <cudaStream_t><size_t>(stream)
+
+    def __init__(self, ptr, stream=0):
+        self.ptr = ptr
+        self.stream = stream
```

Comment:
What is the intended use of this constructor (as opposed to `__cinit__`)?

Reply:
Should probably just make these properties that reference the c attributes

Reply:
Not sure enough your intent here, but we could mark the C attributes as `readonly` so that Python can view them like properties with less boilerplate.

Reply:
Can a `void*` and a `cudaStream_t` be viewed form Python land?

Reply:
Good point. Probably not. Could type them as `uintptr_t` and `size_t` (or whatever the equivalent is here). Though it probably makes more sense to have property methods for these.

Reply:
Sounds good!